internal/linker/link_node_modules.ts

```diff
@@ -148,72 +148,38 @@ async function symlink(target: string, p: string): Promise<boolean> {
 }
 
 /**
- * Resolve to an absolute root node_modules directory.
- * @param root The bazel managed node_modules root such as 'npm/node_modules',  which includes the
- * workspace name as the first segment. May be undefined if there are no third_party node_modules
- * deps.
- * @param startCwd The absolute path that bazel started the action at.
- * @param isExecroot True if the action is run in the execroot, false if the action is run in
- * runfiles root.
- * @param runfiles The runfiles helper object.
- * @return The absolute path on disk where node_modules was installed or if no third party
- * node_modules are deps of the current target the returns the absolute path to
- * `execroot/my_wksp/node_modules`.
+ * Resolve to the path to an external workspace
  */
-async function resolveRoot(
-    root: string|undefined, startCwd: string, isExecroot: boolean, runfiles: Runfiles) {
+async function resolveExternalWorkspacePath(
+    workspace: string, startCwd: string, isExecroot: boolean, execroot: string|undefined,
+    runfiles: Runfiles) {
   if (isExecroot) {
-    // Under execroot, the root will be under an external folder from the startCwd
-    // `execroot/my_wksp`. For example, `execroot/my_wksp/external/npm/node_modules`. If there is no
-    // root, which will be the case if there are no third-party modules dependencies for this
-    // target, set the root to `execroot/my_wksp/node_modules`.
-    return root ? `${startCwd}/external/${root}` : `${startCwd}/node_modules`;
+    // Under execroot, the npm workspace will be under an external folder from the startCwd
+    // `execroot/my_wksp`. For example, `execroot/my_wksp/external/npm`. If there is no
+    // npm workspace, which will be the case if there are no third-party modules dependencies for
+    // this target, npmWorkspace the root to `execroot/my_wksp/node_modules`.
+    return `${execroot}/external/${workspace}`;
```

Comment:
does this interact with `--noenable_legacy_runfiles` in any way?

Reply:
No. The path at the execroot is always `execroot/my_workspace/external/other_workspace`. Bazel doesn't layout an `execroot/other_workspace`